### PR TITLE
Weekly CI: Test all models by ONNX.checker/ORT backend 

### PR DESCRIPTION
**Pull request by jcwchen**
**Description**
- Add a new test script: test all models from ONNX Model Zoo 
- TODO: Complete nightly or weekly testing in GitHub Action (Probably need to divide into test set to avoid memory issue in CIs)
- Refactor the utilities' structure 

**Motivation**
To enhance testing, test all models in ONNX Model Zoo nightly weekly
It is also useful for testing the latest onnx in onnx repo.

### CODE REVIEW

File: .github/workflows/weekly.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Weekly ci workflow for model zoo.
+# Runs model checker for all models
+
+name: Weekly CI with onnx.checker
+
+# Triggers the workflow on push or pull request events but only for the master branch
+on:
+  push:
```
Comment (askhade) [inline .github/workflows/weekly.yml:8]: trigger needs to change right

File: workflow_scripts/test_all_onnx_checker.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            failed_models.append(model_path)
+        end = time.time()
+        print('--------------Time used: {} secs-------------'.format(end - start))
+
+
+    if len(failed_models) == 0:
+        print('{} models have been checked.'.format(len(model_list)))
+    else:
+        print('In all {} models, {} models failed.'.format(len(model_list), len(failed_models)))
+        #sys.exit(1)
```
Comment (askhade) [inline workflow_scripts/test_all_onnx_checker.py:72]: exit with non 0 code is required right?

Comment (jcwchen) [reply workflow_scripts/test_all_onnx_checker.py:72]: Good catch. Thank you

File: workflow_scripts/test_new_models.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+from pathlib import Path
+import subprocess
+from utils import test_utils
+
+
+def main():
+  parser = argparse.ArgumentParser(description='Test settings')
+  # default all: test by both onnx and onnxruntime
+  # if target is specified, only test by the specified one
+  parser.add_argument('--target', required=False, default='all', type=str,
```
Comment (jcwchen) [inline workflow_scripts/test_new_models.py:11]: Duplicate -- `--target` arguments. Not sure whether these two need to be merged together

**Comment by jcwchen** [comment]
Space error can be solved by `git lfs prune` and this PR is ready for review

File: workflow_scripts/utils/check_model.py [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 
-def run_onnx_checker(model_path):
-    model = onnx.load(model_path)
-    onnx.checker.check_model(model)
+def run_onnx_checker(model):
+    # stricter onnx.checker with onnx.shape_inference
+    onnx.checker.check_model(model, True)
 
 def run_backend_ort(model_path, test_data_set=None):
     # if 'test_data_set_N' doesn't exist, create test_dir
```
Comment (jcwchen) [inline workflow_scripts/utils/check_model.py:12]: TODO: resolve conflict -- skip opset<7

**Comment by jcwchen** [comment]
To prevent protobuf error, update broken test_data_set for 3 models includeing vgg19-7, vgg19-bn-7, ResNet101-DUC-7.onnx. 

There are still 73 existing broken test_data_set which cannot pass ORT backend, but these models can pass by creating new test_data_set. Even we do that, there are still 2 models which encounter errors while creating test_data_set: bertsquad-8.onnx, arcfaceresnet100-8. Add them to the skip set now.

Weekly-CI is green now and this PR is ready for review. Thanks.

**Comment by jcwchen** [comment]
Close due to inactivity and we decide not to have such a test in onnx/models. Perhaps doing it in ORT is better.

**Pull request closed without merging**